drivers/timer/intel_adsp_timer.c

```diff
@@ -210,7 +210,9 @@ static void irq_init(void)
 
 void smp_timer_init(void)
 {
+#if !CONFIG_SOF
```

Comment:
Some short note why SOF doesn't want to serve timer irqs on secondary cores, would be nice here for future generations.

Reply:
I think this is wrong to wrap like this. Instead wrap with if !CONFIG_TIMESLICING perhaps

Reply:
I understand that the comment above irq_init() indicates that the timer interrupt is sent to all CPUs for the benefit of timeslicing (added at the end of 2021), but I am not so certain that is relevant anymore. In March of 2023, updates were made to the time slicing infrastructure to base them on timeouts. If the time slice has expired for a thread on a CPU different than the one that serviced the timer ISR, then an IPI is sent which will trigger the scheduler on the other cores.

If there are non-SOF scenarios where we want the timer IRQ to be sent to secondary cores, I am leaning towards keying off a new dedicated Kconfig option for that behavior (it could be enabled/disabled by default by CONFIG_SOF to make configuration compatibility easier). If not, then remove this call to irq_init().

Reply:
I hadn't seen the ipi z_time_slice() addition, nice

Reply:
Yeah, the point was to decouple timer drivers from timeslicing, as every platform has its own rules.